build-scripts/build_xccdf.py

```diff
@@ -88,39 +89,10 @@ def link_ocil(xccdftree, checks, output_file_name, ocil):
     ocil_linker.link_xccdf()
 
 
-def link_benchmark(loader, xccdftree, args, benchmark=None):
-    checks = xccdftree.findall(".//{%s}check" % ssg.constants.XCCDF12_NS)
-
-    link_oval(xccdftree, checks, args.oval, args.build_ovals_dir)
-
-    ocil = loader.export_ocil_to_xml(benchmark)
-    if ocil is not None:
-        link_ocil(xccdftree, checks, args.ocil, ocil)
-
-    ssg.xml.ElementTree.ElementTree(xccdftree).write(args.xccdf)
-
-
-def get_path(path, file_name):
-    return os.path.join(path, file_name)
-
-
-def link_benchmark_per_profile(loader, args):
-    if not os.path.exists(args.thin_ds_components_dir):
-        os.makedirs(args.thin_ds_components_dir)
-
-    loader.off_ocil = True
-
-    for id_, benchmark in loader.get_benchmark_by_profile():
-        xccdftree = benchmark.to_xml_element(loader.env_yaml)
-        p = Paths_(
-            xccdf=get_path(args.thin_ds_components_dir, "xccdf_{}.xml".format(id_)),
-            oval=get_path(args.thin_ds_components_dir, "oval_{}.xml".format(id_)),
-            ocil=get_path(args.thin_ds_components_dir, "ocil_{}.xml".format(id_)),
-            build_ovals_dir=args.build_ovals_dir
-        )
-        link_benchmark(loader, xccdftree, p, benchmark)
-
-    loader.off_ocil = False
+def store_xccdf_per_profile(loader, thin_ds_components_dir):
+    for id_, xccdftree in loader.get_benchmark_xml_by_profile():
+        xccdf_file_name = os.path.join(thin_ds_components_dir, "xccdf_{}.xml".format(id_))
+        ssg.xml.ElementTree.ElementTree(xccdftree).write(xccdf_file_name)
```

Comment:
I think that the write method needs to have the encoding set to "utf-8" because it caused a problem recently, look https://github.com/ComplianceAsCode/content/pull/11614.